### PR TITLE
Update to `dusk-plonk` 0.9

### LINKED ISSUE #165: Update to `dusk-plonk` 0.9
**Describe what you want implemented** 
Update `dusk-plonk` dependency to `0.9`.

**Describe "Why" this is needed**
`dusk-plonk` `0.9` was released with major API improvements and general bug fixes. It is desirable to have these improvements available.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by vlopes11**
Resolves #165

### CODE REVIEW

File: src/tree.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Implementation of a Merkle Tree with a Dusk-Poseidon backend and zero-knowledge opening proof powered by PLONK.
```
Comment (moCello) [inline src/tree.rs:7]: In the previous file 'src/tree/mod.rs' at this point there was a much larger documentation of the tree structure with an example. Is the example no longer valid or is there another reason why you removed?

File: tests/tree.rs [hunk 2]
```diff
@@ -7,17 +7,18 @@
...
     self, PoseidonAnnotation, PoseidonBranch, PoseidonTree,
 };
 use max_annotation::MockLeaf;
 use rand_core::{CryptoRng, OsRng, RngCore};
 
+use dusk_plonk::prelude::*;
+
 const DEPTH: usize = 17;
-const CAPACITY: usize = 1 << 15;
+const CAPACITY: usize = 15;
```
Comment (moCello) [inline tests/tree.rs:21]: Are you sure the capacity is 15? Seems a significant change from `1 << 15`

**Review by moCello — APPROVED** [review]
I left one question but other than that it looks good to me

File: src/tree.rs (hunk 1 above)
Comment (vlopes11) [reply src/tree.rs:7]: It is duplicated with the `README.md`

**Pull request merged**